### ISSUE TITLE
manual_range_contains suggests breaking code constness

### ISSUE THREAD

**Issue by tspiteri** [rust-lang/rust-clippy#6373]
```rust
pub const fn in_range(a: i32) -> bool {
    3 <= a && a <= 20
}
```

Clippy says
```
warning: manual `RangeInclusive::contains` implementation
 --> src/lib.rs:2:5
  |
2 |     3 <= a && a <= 20
  |     ^^^^^^^^^^^^^^^^^ help: use: `(3..=20).contains(&a)`
  |
  = note: `#[warn(clippy::manual_range_contains)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_range_contains
```

But that would make the code fail to compile because `error[E0015]: calls in constant functions are limited to constant functions, tuple structs and tuple variants`.

### Meta

Clippy: 0.0.212 (2020-11-23 d9a105f)

**Comment by giraffate** [comment]
I will prepare a fix.
